letsencrypt/configuration.py

```diff
@@ -82,6 +84,21 @@ def http01_port(self):  # pylint: disable=missing-docstring
         else:
             return challenges.HTTP01Response.PORT
 
+    def determine_user_agent(self, authenticator, installer):
```

Comment:
I think you're probably right, that would be more elegant.  Not sure if we should block on it, though...